snuba/query/parser/expressions.py

```diff
@@ -1,28 +1,181 @@
-from typing import Any, Optional
+from dataclasses import replace
+from typing import List, Tuple, Union
 
-from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from parsimonious.grammar import Grammar  # type: ignore
+from parsimonious.nodes import Node, NodeVisitor  # type: ignore
+from typing import Any, Iterable, Optional
+
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
 from snuba.query.parser.functions import parse_function_to_expr
 from snuba.util import is_function, QUOTED_LITERAL_RE
 
+minimal_clickhouse_grammar = Grammar(
+    r"""
+# This root element is needed because of the ambiguity of the aggregation
+# function field which can mean a clickhouse function expression or the simple
+# name of a clickhouse function.
+root_element    = function_call / function_name
+expression      = function_call / simple_term
+parameters_list = parameter* (expression)
+parameter       = expression space* comma space*
+function_call   = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)?
+simple_term     = quoted_literal / numeric_literal / column_name
+literal         = ~r"[a-zA-Z0-9_\.:-]+"
+quoted_literal  = "'" string_literal "'"
+string_literal  = ~r"[a-zA-Z0-9_\.:-]+"
+numeric_literal = ~r"-?[0-9]+(\.[0-9]+)?"
+column_name     = ~r"[a-zA-Z_][a-zA-Z0-9_\.]*"
+function_name   = ~r"[a-zA-Z_][a-zA-Z0-9_]*"
+open_paren      = "("
+close_paren     = ")"
+space           = " "
+comma           = ","
+"""
```

Comment:
This seems like a good example for if/when we support additional input formats. 👌

Reply:
Well, technically parsimonious does not like a rule formatted as 
> token "(" token "," token....
seems you cannot interleave tokens and literals, so we do not have an alternative.

Reply:
(For clarification, I meant this as a block comment from L18-L39. I think if we want to have a textual input format this would be a good way to support it.)

Reply:
I'll add something after verifying this works properly.